kernel/src/hil/servo.rs

```diff
@@ -0,0 +1,25 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2024.
+
+use crate::ErrorCode;
+pub trait Servo<'a> {
+    /// Changes the angle of the servo.
+    /// Return values:
+    ///
+    /// - `Ok(())`: The attempt at changing the angle was successful.
+    /// - `FAIL`: Cannot change the angle.
+    /// - `INVAL`: The value exceeds u16, indicating it's incorrect
+    /// since servomotors can only have a maximum of 360 degrees.
+    ///  # Arguments
+    /// - `angle` - the variable that receives the angle
+    /// (in degrees from 0 to 180) from the servo driver.
+    fn set_angle(&self, angle: u16) -> Result<(), ErrorCode>;
+
+    /// Returns the angle of the servo.
+    /// Return values:
+    ///
+    /// - `angle`: The value, in angles from 0 to 360, of the servo.
+    /// - `OFF`:  The servo was not set to an angle.
```

Comment:
Please document that this should return `NOSUPPORT` if the servo is not able to report the angle.